pkg/reconciler/apibinding/apibinding_reconcile_test.go

```diff
@@ -0,0 +1,745 @@
+				"someresources": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "someresources",
+						UID:  "uid1",
+					},
+				},
+				"moreresources": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "moreresources",
+						UID:  "uid3",
+					},
+				},
+			},
+			wantBinding:         true,
+			wantReconcileStatus: reconcileStatusStop,
+		},
+		"bound becomes binding when bound APIResourceSchema UID changes": {
+			apiBinding: bound.Build(),
+			apiExport: &apisv1alpha1.APIExport{
+				Spec: apisv1alpha1.APIExportSpec{
+					LatestResourceSchemas: []string{"someresources", "otherresources"},
+				},
+			},
+			apiResourceSchemas: map[string]*apisv1alpha1.APIResourceSchema{
+				"someresources": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "someresources",
+						UID:  "uid1",
+					},
+				},
+				"otherresources": {
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "otherresources",
+						UID:  "newuid",
+					},
+				},
+			},
+			wantBinding:         true,
+			wantReconcileStatus: reconcileStatusStop,
+		},
+		"APIExportValid warning condition set when error getting previously bound APIExport": {
+			apiBinding:          bound.Build(),
+			getAPIExportError:   errors.New("foo"),
+			wantError:           true,
+			wantReconcileStatus: reconcileStatusStop,
+			wantCondition: &wantCondition{
+				Type:     apisv1alpha1.APIExportValid,
+				Status:   corev1.ConditionFalse,
+				Reason:   apisv1alpha1.GetErrorReason,
+				Severity: conditionsv1alpha1.ConditionSeverityWarning,
+			},
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			r := &phaseReconciler{
+				getAPIExport: func(clusterName, name string) (*apisv1alpha1.APIExport, error) {
+					require.Equal(t, "org:some-workspace", clusterName)
+					require.Equal(t, "some-export", name)
+					return tc.apiExport, tc.getAPIExportError
+				},
+				getAPIResourceSchema: func(clusterName, name string) (*apisv1alpha1.APIResourceSchema, error) {
+					require.Equal(t, "org:some-workspace", clusterName)
+					return tc.apiResourceSchemas[name], nil
+				},
+			}
+
+			status, err := r.reconcile(context.Background(), tc.apiBinding)
+
+			gotErr := err != nil
+			require.Equal(t, tc.wantError, gotErr)
+			if tc.wantError {
+				return
+			}
+
+			require.Equal(t, status, tc.wantReconcileStatus)
+
+			switch {
+			case tc.wantBinding:
+				require.Equal(t, apisv1alpha1.APIBindingPhaseBinding, tc.apiBinding.Status.Phase)
+			case tc.wantBound:
+				require.Equal(t, apisv1alpha1.APIBindingPhaseBound, tc.apiBinding.Status.Phase)
+			}
+
+			if tc.wantCondition != nil {
+				requireConditionMatches(t, tc.apiBinding, *tc.wantCondition)
+			}
+		})
+	}
+}
+
+func TestCRDFromAPIResourceSchema(t *testing.T) {
+	tests := map[string]struct {
+		schema  *apisv1alpha1.APIResourceSchema
+		want    *apiextensionsv1.CustomResourceDefinition
+		wantErr bool
+	}{
+		"full schema": {
+			schema: &apisv1alpha1.APIResourceSchema{
```

Comment:
(No action required) Given that this is a unit test and the crd doesn't have to hit apiserver validation, maybe limit the fields defined in tests cases to those that the function under test actually cares about?

Reply:
here it is important: we want to see that the CRD really matches